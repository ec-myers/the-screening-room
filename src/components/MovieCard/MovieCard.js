import React from 'react';
import './MovieCard.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { toggleFavorite } from '../../actions';

export const MovieCard = ({
  movie,
  favorites,
  toggleFavorite,
  updateFavorites
}) => {
  const { title, poster, favorite, id } = movie;
  const isFavorite = favorites
    .map(favorite => favorite.title)
    .includes(movie.title);

  const handleClick = () => {
    toggleFavorite(title);
    updateFavorites(movie, isFavorite);
  };
  return (
    <div className='MovieCard__div--container'>
      <Link to={`movies/${id}`}>
        <img
          className='MovieCard__img--poster'
          src={poster}
          alt='Official movie poster'
        />
      </Link>
      <footer className={`MovieCard__footer ${favorite ? 'footer-active' : ''}`}>
        <h3 className='MovieCard__h3--title'>{title}</h3>
        <div className={`button-container ${favorite ? 'button-container-active' : ''}`} onClick={handleClick}>
          <div className={favorite ? 'stick-1-active' : 'stick-1'}></div>
          <div className={favorite ? 'stick-2-active' : 'stick-2'}></div>
        </div>
      </footer>
    </div>
  );
};

export const mapStateToProps = ({ movies, favorites, user }) => ({
  movies,
  favorites,
  user
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ toggleFavorite }, dispatch);
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieCard);
