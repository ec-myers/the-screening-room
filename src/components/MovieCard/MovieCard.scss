.MovieCard__div--container {
  align-items: center;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.301);
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 2% 1%;
  width: 350px;
  transition: .15s;
  cursor: pointer;

  &:hover {
    box-shadow: 0 2px 20px rgba(0, 0, 0, 0.6);
    transition: .15s;
  }
}

.MovieCard__img--poster {
  height: 525px;
  width: 350px;
}

.MovieCard__footer {
  background: linear-gradient(
    -0.25turn,
    rgb(3, 190, 103),
    rgb(3, 204, 110),
    rgb(25, 211, 124),
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white,
    white
  );
  text-align: center;
  height: 60px;
  margin: 0%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0%;
  width: 100%;
}

.MovieCard__h3--title {
  margin: 0;
  margin-left: 20px;
  margin-top: 2%;
  padding: 0;
  width: 75%;
  text-align: left;
  padding-bottom: 6px;
}

.button-container {
  width: 45px;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  div {
    width: 4px;
    border-radius: 1px;
  }
}

.stick-1 {
  transform: rotate(90deg);
  height: 22px;
  background-color: rgba(255, 255, 255, 0.918);
  position: absolute;
  z-index: 0;
  transition: .2s;
}

.stick-2 {
  background-color: rgb(255, 255, 255);
  height: 22px;
  transition: .2s;
}

.stick-1-active {
  transform: rotate(40deg);
  height: 23px;
  background-color: rgb(255, 255, 255);
  position: absolute;
  margin-left: 5px;
  transition: .2s;
  border-radius: 0px;
}

.stick-2-active {
  height: 8px;
  margin-right: 11px;
  background-color: rgba(255, 255, 255, 0.911);
  transform: rotate(-49deg);
  margin-top: 7px;
  transition: .2s;
  border-radius: 0px;
}