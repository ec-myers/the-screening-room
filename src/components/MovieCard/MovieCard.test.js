import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard, mapStateToProps, mapDispatchToProps } from './MovieCard';

describe('MovieCard', () => {
  let wrapper;
  let toggleFavorites = jest.fn();
  let movie = {
    favorite: false,
    genre: (3)[('Crime', 'Drama', 'Thriller')],
    id: 475557,
    overview:
      'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.',
    poster:
      'https://image.tmdb.org/t/p/original/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg',
    rRated: false,
    rating: 8.6,
    release_date: '2019-10-04',
    title: 'Joker',
    wallpaper:
      'https://image.tmdb.org/t/p/original/n6bUvigpRFqSwmPp1m2YADdbRBc.jpg'
  };

  beforeEach(() => {
    wrapper = shallow(
      <MovieCard {...movie} key={movie.id} toggleFavorites={toggleFavorites} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
