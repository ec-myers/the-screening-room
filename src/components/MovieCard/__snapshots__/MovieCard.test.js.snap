// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<div
  className="MovieCard__div--container"
>
  <Link
    to="movies/475557"
  >
    <img
      alt="Official movie poster"
      className="MovieCard__img--poster"
      src="https://image.tmdb.org/t/p/original/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
    />
  </Link>
  <footer
    className="MovieCard__footer"
  >
    <h3
      className="MovieCard__h3--title"
    >
      Joker
    </h3>
    <div
      className="button-container"
      onClick={[Function]}
    >
      <div
        className="stick-1"
      />
      <div
        className="stick-2"
      />
    </div>
  </footer>
</div>
`;
